docs/source/engine.rst

```diff
@@ -133,13 +133,13 @@ Deterministic training
 ``````````````````````
 
 In general, it is rather difficult task to achieve deterministic and reproducible trainings as it relies on multiple
-aspects, e.g. data version, code version, software environment, hardware etc. According to `PyTorch documentation <https://pytorch.org/docs/stable/notes/randomness.html>`_:
+aspects, e.g. data version, code version, software environment, hardware etc. According to `PyTorch documentation <https://pytorch.org/docs/stable/notes/randomness.html>`__:
```

Comment:
AFAIK, a link in RST is done as 
```
`name <url>`_
```
For example: https://docutils.sourceforge.io/docs/user/rst/quickref.html#hyperlink-targets

Why do we need to add a dunder ?

Reply:
I found this warning when building docs, found to use [anonymous-hyperlinks](https://docutils.sourceforge.io/docs/ref/rst/restructuredtext.html#anonymous-hyperlinks) to solve from [stackoverflow](https://stackoverflow.com/questions/5464627/how-to-have-same-text-in-two-links-with-restructured-text)
```py
WARNING: duplicate target name: "pytorch documentation"
```